### PR TITLE
add expand as op , fix some bugs

### PR DESCRIPTION
**Pull request by JesseyXujin**
more details see this pr:https://github.com/PaddlePaddle/Paddle/pull/19986/files
support targed tensor input.

### CODE REVIEW

File: paddle/fluid/operators/expand_as_op.cu [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline paddle/fluid/operators/expand_as_op.cu]: 2016 -> 2019

**Review by xyzhou-puck — DISMISSED** [review]
LGTM.

**Review by xyzhou-puck — APPROVED** [review]
LGTM.

**Review by heavengate — APPROVED** [review]

**Review by kuke — APPROVED** [review]
LGTM

**Pull request closed without merging**